lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,384 @@
+          else
+            check_ssh_error ssh_output
+            exit 1
+          end
+        end
+
+        def add_ip_to_known_hosts
+          jump_box_ip = `grep -A 2 'Host socks' ~/.ssh/config | grep ProxyCommand | awk '{print $6}'`.chomp
+          socks_ip = `grep -A 2 'Host socks' ~/.ssh/config | grep HostName | awk '{print $2}'`.chomp
+
+          `ssh-keygen -R #{jump_box_ip}` if File.exist? '~/.ssh/known_hosts'
+          `ssh-keygen -R #{socks_ip}` if File.exist? '~/.ssh/known_hosts'
+          `ssh-keyscan -H #{jump_box_ip} >> ~/.ssh/known_hosts 2> /dev/null`
+          `ssh -i #{ssh_key_path} dsva@#{jump_box_ip} 'ssh-keyscan -H #{socks_ip}' >> ~/.ssh/known_hosts 2> /dev/null`
+        end
+
+        def check_ssh_error(ssh_output)
+          if ssh_output.include? 'Permission denied (publickey)'
+            @skip_test = true
+            output.puts '⚠️  WARN: SSH key is not approved yet. Once it is, re-run `vtk socks setup`.'
+          else
+            ssh_command = "ssh -i #{ssh_key_path} -F #{ssh_config_path} -o ConnectTimeout=5 -vvv socks -D #{port} -N"
+            output.puts ' ❌ ERROR: SSH Connection to SOCKS server unsuccessful. Error message:'
+            output.puts ssh_command
+            output.puts `#{ssh_command}`
+          end
+        end
+
+        def configure_system_boot
+          return false unless macos?
+
+          la_output = `launchctl list | grep #{launch_agent_label}`
+          return true if !la_output.empty? && !la_output.start_with?('-')
+
+          log 'Configuring SOCKS tunnel to run on system boot...' do
+            install_autossh && install_launch_agent
+          end
+        end
+
+        def launch_agent_label
+          @launch_agent_label ||= begin
+            launch_agent_label = 'gov.va.socks'
+            launch_agent_label += "-test-#{rand 1000}" if ENV['TEST'] == 'test'
+            launch_agent_label
+          end
+        end
+
+        def install_autossh
+          installed = !`command -v autossh`.empty?
+          return true if installed
+
+          system 'brew install autossh'
+        end
+
+        def install_launch_agent
+          unless File.exist? "#{boot_script_path}/LaunchAgents/gov.va.socks.plist"
+            FileUtils.mkdir_p "#{boot_script_path}/Logs/autossh"
+            FileUtils.mkdir_p "#{boot_script_path}/LaunchAgents"
+
+            write_launch_agent
+          end
+
+          system "launchctl unload #{boot_script_path}/LaunchAgents/gov.va.socks.plist 2> /dev/null"
+          system "launchctl load -w #{boot_script_path}/LaunchAgents/gov.va.socks.plist"
+        end
+
+        def write_launch_agent
+          erb_template = File.read "#{__dir__}/gov.va.socks.plist.erb"
+          erb = ERB.new erb_template
+          launch_agent_contents = erb.result(
+            launch_agent_variables.instance_eval { binding }
+          )
+          File.write "#{boot_script_path}/LaunchAgents/gov.va.socks.plist", launch_agent_contents
```

Comment:
Something like this should work:
```ruby
TTY::File.copy_file("templates/gov.va.socks.plist.erb", "#{boot_script_path}/LaunchAgents/", context: launch_agent_variables)
```

https://www.rubydoc.info/gems/tty-file#2-4-copy_file